pkg/apis/contrail/v1alpha1/zookeeper_types.go

```diff
@@ -82,88 +79,59 @@ func init() {
-func (c *Zookeeper) InstanceConfiguration(request reconcile.Request,
+func (c *Zookeeper) InstanceConfiguration(request reconcile.Request, confCMName string,
 	podList *corev1.PodList,
 	client client.Client) error {
-	instanceConfigMapName := request.Name + "-" + "zookeeper" + "-configmap"
-	configMapInstanceDynamicConfig := &corev1.ConfigMap{}
-	err := client.Get(context.TODO(),
-		types.NamespacedName{Name: instanceConfigMapName, Namespace: request.Namespace},
-		configMapInstanceDynamicConfig)
+	zookeeperConfigInterface := c.ConfigurationParameters()
+	zookeeperConfig := zookeeperConfigInterface.(ZookeeperConfiguration)
+	pods := make([]corev1.Pod, len(podList.Items))
+	copy(pods, podList.Items)
+	sort.SliceStable(pods, func(i, j int) bool { return pods[i].Name < pods[j].Name })
+
+	confCMData, err := configtemplates.DynamicZookeeperConfig(podList, strconv.Itoa(*zookeeperConfig.ElectionPort), strconv.Itoa(*zookeeperConfig.ServerPort))
 	if err != nil {
 		return err
 	}
-	configMapInstancConfig := &corev1.ConfigMap{}
-	err = client.Get(context.TODO(),
-		types.NamespacedName{Name: instanceConfigMapName + "-1", Namespace: request.Namespace},
-		configMapInstancConfig)
+
+	var zookeeperLogBuffer, zookeeperXslBuffer bytes.Buffer
+	err = configtemplates.ZookeeperLogConfig.Execute(&zookeeperLogBuffer, struct{}{})
 	if err != nil {
 		return err
 	}
-
-	zookeeperConfigInterface := c.ConfigurationParameters()
-	zookeeperConfig := zookeeperConfigInterface.(ZookeeperConfiguration)
-	sort.SliceStable(podList.Items, func(i, j int) bool { return podList.Items[i].Status.PodIP < podList.Items[j].Status.PodIP })
-	for _, pod := range podList.Items {
-		myidString := pod.Name[len(pod.Name)-1:]
-		myidInt, err := strconv.Atoi(myidString)
-		if err != nil {
-			return err
-		}
-		if configMapInstanceDynamicConfig.Data == nil {
-			data := map[string]string{pod.Status.PodIP: strconv.Itoa(myidInt + 1)}
-			configMapInstanceDynamicConfig.Data = data
-		} else {
-			configMapInstanceDynamicConfig.Data[pod.Status.PodIP] = strconv.Itoa(myidInt + 1)
-		}
-		var zkServerString string
-		for _, pod := range podList.Items {
-			myidString := pod.Name[len(pod.Name)-1:]
-			myidInt, err := strconv.Atoi(myidString)
-			if err != nil {
-				return err
-			}
-			zkServerString = zkServerString + fmt.Sprintf("server.%d=%s:%s:participant\n",
-				myidInt+1, pod.Status.PodIP,
-				strconv.Itoa(*zookeeperConfig.ElectionPort)+":"+strconv.Itoa(*zookeeperConfig.ServerPort))
-		}
-		configMapInstanceDynamicConfig.Data["zoo.cfg.dynamic.100000000"] = zkServerString
-		err = client.Update(context.TODO(), configMapInstanceDynamicConfig)
-		if err != nil {
-			return err
-		}
-		var zookeeperConfigBuffer, zookeeperLogBuffer, zookeeperXslBuffer bytes.Buffer
-
-		err = configtemplates.ZookeeperConfig.Execute(&zookeeperConfigBuffer, struct {
-			ClientPort string
+	err = configtemplates.ZookeeperXslConfig.Execute(&zookeeperXslBuffer, struct{}{})
+	if err != nil {
```

Comment:
Nit: We can combine this err check with previous statement.

Reply:
it's not my code, i'm removing it in the next PR